app/server/lib/FlexServer.ts

```diff
@@ -1883,6 +1883,19 @@ export class FlexServer implements GristServer {
     const probes = new BootProbes(this.app, this, '/api', adminMiddleware);
     probes.addEndpoints();
 
+    this.app.post('/api/admin/restart', requireInstallAdmin, expressWrap(async (req, resp) => {
+      const newEnv = req.body.newEnv;
```

Comment:
Hmmm so anything sent to the endpoint gets injected in the environment? That is very general purpose. I'm trying to think if there's any mischief that could be caused my setting unexpected variables, like HOME or USER. I guess it just becomes the case that an admin has this power, they'd need to have it to e.g. set the database to a different url. Scary but makes sense.

My other question is what is the persistence story? For this endpoint, how long are the settings supposed to stick? In current implementation, they'd evaporate if the overall container is restarted, right?

Reply:
For the security endpoint, I assume that if an admin wants to specifically craft a new environment and restart, that's on them. The endpoint is already limited to admins, so I think that's okay, as you say.

For persistence, as we discussed elsewhere, this already seems to be handled by the existing code: once an activation key is present, [it's saved into the database](https://github.com/gristlabs/grist-ee/blob/main/ext/app/server/lib/ActivationReader.ts#L89). We just need to make sure that current code that we write now is aware of the presence of this key.